modules/core/arrow-validation/src/main/kotlin/arrow/validation/refinedTypes/NonZero.kt

```diff
@@ -0,0 +1,47 @@
+package arrow.validation.refinedTypes
+
+import arrow.Kind
+import arrow.core.Either
+import arrow.core.EitherPartialOf
+import arrow.data.Nel
+import arrow.data.Validated
+import arrow.data.ValidatedPartialOf
+import arrow.extension
+import arrow.instances.either.applicativeError.applicativeError
+import arrow.instances.nonemptylist.semigroup.semigroup
+import arrow.instances.validated.applicativeError.applicativeError
+import arrow.typeclasses.ApplicativeError
+import arrow.validation.RefinedPredicateException
+import arrow.validation.Refinement
+
+interface NonZero<F, A : Number> : Refinement<F, A> {
+
+  override fun A.refinement(): Boolean = this != 0
+
+  fun A.nonZero(): Kind<F, A> = refine(this)
```

Comment:
If I add a method `fun nonZero(a: A): Kind<F, A> = a.nonZero()` the IDE tells me its signature clashes with this method 🤔